source/common/upstream/upstream_impl.cc

```diff
@@ -1088,6 +1093,17 @@ void ClusterImplBase::validateEndpointsForZoneAwareRouting(
   }
 }
 
+ClusterInfoImpl::OptionalClusterStats::OptionalClusterStats(
+    const envoy::config::cluster::v3::Cluster& config, Stats::Scope& stats_scope)
+    : timeout_budget_stats_(
+          (config.track_cluster_stats().timeout_budgets() or config.track_timeout_budgets())
```

Comment:
s/or/||/

Reply:
so if at least one of flags is set to true, histograms will be enabled. We may need to document this expectation or add validation that if `config.track_cluster_stats().timeout_budgets() != config.track_timeout_budgets()` then fail on startup.

Reply:
Makes sense. Not sure which one is preferred while deprecating such fields.

Reply:
after consulting with @htuch, is acceptable to make new field override the old one. The way is implemented now fits this strategy (newer one take priority).